### ISSUE TITLE
SA1137: Unexpected behaviour with switch statements

### ISSUE THREAD

**Issue by berkeleybross** [DotNetAnalyzers/StyleCopAnalyzers#2527]
Suppose I have a method with a switch statement in it:

```
private string ToLongHand(int x)
{
    switch (x)
    {
        case 1:
            return "One";
        case 2:
        {
            // Do some complicated logic
            return "Two";
        }

        default:
            throw new ArgumentOutOfRangeException(nameof(x), "I cant count past 2...");
    }
}
```

SA1137 is raised for the brace immediately after case 2. Is that intended? Whilst personally I think this code is acceptable, I might be persuaded by a rule saying that all cases should have a brace or none of them should have.

Additionally, when I run the fix, it converts the code to the following:

```
private string ToLongHand(int x)
{
    switch (x)
    {
        case 1:
            return "One";
        case 2:
            {
            // Do some complicated logic
            return "Two";
        }

        default:
            throw new ArgumentOutOfRangeException(nameof(x), "I cant count past 2...");
    }
}
```

... which is definitely worse than before!

Theres also an issue if the first case statement has a brace and the second one doesnt - it highlights the second block and suggests that it should be aligned with the start of the "case" statement

I'm using StyleCop.Analyzers.1.1.0-beta004 on Visual Studio 2017 (v15.3.2)

**Comment by sharwell** [comment]
Yes, this is a currently-unhandled edge case. I would expect the correct solution would involve comparing switch cases that contain exactly one block as a child to each other, and all other cases the way they are currently compared.
